### PR TITLE
Add zap integration

### PR DESCRIPTION
**Pull request by chimanjain**
Add in [zap](https://github.com/uber-go/zap) support for Gobrake.

### CODE REVIEW

File: examples/zap/main.go [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+	ProjectKey:  ProjectKey,
+	Environment: "production",
+})
+
+/*
+Note: This example only shows how to send errors to Airbrake
+i.e. how to create a core and send the logs to airbrake
+If you want to write the logs to json file or stdout
+you should use zapcore.NewTee() for using multiple cores
+You can set the severity accordingly
```
Comment (kyrylo) [inline examples/zap/main.go:27]:
```suggestion
Note: This example only shows how to send errors to Airbrake,
i.e., how to create a core and send the logs to Airbrake.
If you want to write the logs to a JSON file or stdout,
you should use `zapcore.NewTee()` for using multiple cores.
You can set the severity accordingly.
```

File: examples/zap/main.go [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+*/
+func main() {
+	zapbrake, err := zapbrake.NewCore(zapcore.ErrorLevel, notifier)
+	if err != nil {
+		// The only way this error would be returned is if airbrake was not set up and passed in as a nil value
+		// Either stop the execution of the code or ignore it, as zapbrake is set to an empty core that wont write to
+		// airbrake if a pointer to airbrake notifier is not provided.
```
Comment (kyrylo) [inline examples/zap/main.go:34]:
```suggestion
		// The only case when this error would be returned is when Airbrake was not set up and passed in as a nil
		// value. Either stop the execution of the code or ignore it, as zapbrake is set to an empty core that won't write to
		// Airbrake if a pointer to the Airbrake notifier is not provided.
```

File: zap/zap.go [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+type Core struct {
+	zapcore.LevelEnabler
+
+	Notifier *gobrake.Notifier
+
+	coreFields map[string]interface{}
+	depth      int
+}
+
+func NewCore(enab zapcore.LevelEnabler, notifier *gobrake.Notifier, options ...Option) (*Core, error) {
```
Comment (kyrylo) [inline zap/zap.go:21]: NewCore accepts options. Can you, please, provide an example in `examples/zap` about how to pass these options?

Comment (chimanjain) [reply zap/zap.go:21]: I will be removing the `options` as I don't see it's necessity for any extraordinarily configuration.

File: examples/zap/main.go (hunk 1 above)
Comment (chimanjain) [reply examples/zap/main.go:27]: Do we need commas before and after `i.e`?

Comment (kyrylo) [reply examples/zap/main.go:27]: https://jakubmarian.com/comma-after-i-e-and-e-g/

**Comment by chimanjain** [comment]
PTAL

File: zap/zap.go [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+	return core, nil
+}
+
+// SetDepth method is for setting the depth of the notices
+func (core *Core) SetDepth(depth int) {
+	core.depth = depth
+}
+
+func (core *Core) With(fields []zapcore.Field) zapcore.Core {
+	clone := *core
```
Comment (kyrylo) [inline zap/zap.go:35]: Why do we need to make a shallow copy and return it?

Comment (chimanjain) [reply zap/zap.go:35]: I will remove it

File: zap/zap.go [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+}
+
+func (core *Core) Check(entry zapcore.Entry, checked *zapcore.CheckedEntry) *zapcore.CheckedEntry {
+	if core.Enabled(entry.Level) {
+		return checked.AddCore(entry, core)
+	}
+	return checked
+}
+
+func (core *Core) Sync() error {
```
Comment (kyrylo) [inline zap/zap.go:83]: Why is this needed?

Comment (chimanjain) [reply zap/zap.go:83]: Every NewCore requires 4 mandatory methods i.e. With(), Write(), Check(), Sync().

Comment (kyrylo) [reply zap/zap.go:83]: Shouldn't it flush the buffer?

Comment (chimanjain) [reply zap/zap.go:83]: I don't think it has to flush the buffer as we are not using `bufio.Writer`.

**Review by kyrylo — APPROVED** [review]

File: examples/zap/main.go [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+		zap.String("file", "something.png"),
+		zap.String("type", "image/png"),
+		zap.String("user", "tobi"),
+	)
+	logger.Info("upload")          // This log is not sent to Airbrake because the log level is set to Error
+	logger.Info("upload complete") // This log is not sent to Airbrake because the log level is set to Error
+	logger.Warn("upload retry")    // This log is not sent to Airbrake because the log level is set to Error
+
+	logger.Error("upload failed") // This log is sent to Airbrake because the log level is set to Error
+	logger.Sync()
```
Comment (kyrylo) [inline examples/zap/main.go:60]: Maybe no need to call this in the example? It doesn't do anything.

Comment (chimanjain) [reply examples/zap/main.go:60]: In this case it will do nothing, but when we are using it inside zapcore.NewTee() to use multiple core and they have to use this function when they are calling filewriter or terminal writer. They included this in their standard logging example https://pkg.go.dev/go.uber.org/zap#example-package-Presets.
